source/NVDAObjects/UIA/__init__.py

```diff
@@ -1647,6 +1647,11 @@ def _get_states(self):
 		if s!=UIAHandler.handler.reservedNotSupportedValue:
 			if not role:
 				role=self.role
+			if s == UIAHandler.ToggleState_Indeterminate:
+				if role == controlTypes.Role.TOGGLEBUTTON:
+					states.add(controlTypes.State.HALFPRESSED)
+				else:

```

Comment:
Shouldn't this explicitly check if this is a checkbox?

```suggestion
				elif role == cpmtrp;Types.Role.CHECKBOX:
```

Reply:
There are checkable tree view items, so just in case, if it has a toggle pattern then the state is added